aten/src/ATen/native/Math.h

```diff
@@ -267,6 +348,23 @@ static inline float calc_digamma(float x) {
   return result + logf(x) - (0.5f / x) - y;
 }
 
+static inline double calc_polygamma(int64_t n, double x) {
```

Comment:
Could these two functions be implemented using a simple template? Given that it is called from dispatch floating point types where you have `scalar_t` to template over.

Reply:
I think your right... I'll change this(and digamma) too

Reply:
(I don't know why) all functions in Aten/Math.h do not use template at all, so I followed that. let me know if it's required to fix all functions to use template, I'll fix every function in Math.h

Reply:
Ok if it matches the other functions in the file, it's fine.
If you want to send a follow up cleaning that up, please do so ! We're always happy to clean up code! But maybe in a separate PR just to make reviewing easier.